src/Test.EndToEnd.Baselining/Extensions/ArtifactUriEnumeration.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Sarif;
+
+namespace SarifBaseline.Extensions
+{
+    public static class ArtifactUriEnumeration
+    {
+        public static IEnumerable<Uri> AllArtifactUris(this SarifLog log)
+        {
+            if (log != null)
+            {
+                foreach (Run run in log.EnumerateRuns())
+                {
+                    foreach (Uri uri in run.AllArtifactUris())
+                    {
+                        yield return uri;
+                    }
+                }
+            }
+        }
+
+        public static IEnumerable<Uri> AllArtifactUris(this Run run)
+        {
+            if (run?.Artifacts != null)
+            {
+                foreach (Artifact artifact in run.Artifacts)
+                {
+                    Uri uri = artifact?.Location?.Uri;
+                    if (uri != null) { yield return uri; }
+                }
+            }
+
+            foreach (Result result in run.EnumerateResults())
+            {
+                foreach (Uri uri in AllArtifactUris(result))
+                {
+                    yield return uri;
+                }
+            }
+        }
+
+        public static IEnumerable<Uri> AllArtifactUris(this Result result)
+        {
+            if (result?.Locations != null)
+            {
+                foreach (Location location in result.Locations)
+                {
+                    Uri uri = location?.PhysicalLocation?.ArtifactLocation?.FileUri(result.Run);
+                    if (uri != null) { yield return uri; }
+                }
+            }
```

Comment:
>} [](start = 12, length = 1)

To emphasize my earlier point: `result` also has `codeFlows` and `stacks` properties (probably others) which ultimately contain `artifactLocation` objects.